rio_tiler/utils.py

```diff
@@ -161,6 +162,24 @@ def get_vrt_transform(
     dst_transform, _, _ = calculate_default_transform(
         src_dst.crs, dst_crs, src_dst.width, src_dst.height, *src_dst.bounds
     )
+
+    # If bounds window is aligned with the dataset internal tile we align the bounds with the pixels.
+    # This is to limit the number of internal block fetched.
+    if _requested_tile_aligned_with_internal_tile(
+        src_dst, bounds, height, width, dst_crs
+    ):
+        col_off, row_off, w, h = windows.from_bounds(
+            *bounds, transform=src_dst.transform, width=width, height=height,
+        ).flatten()
+
+        w = windows.Window(
+            round(col_off, window_precision),
+            round(row_off, window_precision),
+            round(w, window_precision),
+            round(h, window_precision),
+        )
+        bounds = src_dst.window_bounds(w)
```

Comment:
If dataset is tiled and the window defined by the bbox is aligned with the internal tile then we define update the bounds to match the one defined by the internal block. We do this by using the `round` of the window
```python
Window(
    col_off=8192.000000016182,
    row_off=12288.000000016851,
    width=2047.999999999971, 
    height=2047.999999999971
)
Window(col_off=8192.0, row_off=12288.0, width=2048.0, height=2048.0)
```